pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -943,6 +939,48 @@ func (r *ReconcileClusterDeployment) reconcileCompletedProvision(cd *hivev1.Clus
 }
 
 func (r *ReconcileClusterDeployment) clearOutCurrentProvision(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) (reconcile.Result, error) {
+	// clear out current provision only if install restart is required
+	if cd.Status.InstallRestarts == 0 && cd.Annotations[tryInstallOnceAnnotation] == "true" {
+		cdLog.Debug("not creating new provision since the deployment is set to try install only once")
+		return reconcile.Result{}, nil
+	}
+	if cd.Spec.InstallRestartLimit != nil && cd.Status.InstallRestarts >= *cd.Spec.InstallRestartLimit {
+		cdLog.Debug("not creating new provision since the install restarts limit has been reached")
+		conditions, changed := controllerutils.SetClusterDeploymentConditionWithChangeCheck(
+			cd.Status.Conditions,
+			hivev1.ProvisionStoppedCondition,
+			corev1.ConditionTrue,
+			"InstallRestartsLimitReached",
+			"Install restarts limit reached",
```

Comment:
```suggestion
			"InstallRestartLimitReached",
			"Install restart limit reached",
```
or change the name of the field to `InstallRestartsLimit`.